src/System.Private.CoreLib/src/System/AppContext.Unix.cs

```diff
@@ -17,7 +19,34 @@ public static string BaseDirectory
         {
```

Comment:
I was thinking about this too but was not sure how frequently this API is usually called given the existing Windows implementation. Since you confirmed it then I will add a cache for the value of this property.